### PR TITLE
Add a dataset export button for tasks

### PR DESCRIPTION
**Pull request by zhiltsov-max**
Extends #813 by adding an UI element to export tasks as datasets.

Added:
- REST API call `server/annotations/dataset_export_formats`
- A button "Export as Dataset" for tasks in the dashboard

### CODE REVIEW

**Comment by nmanovic** [comment]
@zhiltsov-max , a couple of comments:

- When I try to export a task it redirect me to the browsable REST API page

image

- I really don't think that users want to select a specific COCO format. Can we select "COCO JSON 1.0" format and dump annotations in all possible ways (labels.json, captions.json, ...)? Do you think it is an acceptable solution? If for a sub-format we don't have annotations probably it is a good sign to skip it.


- The main point of the previous item is to have the same set of format files as we use for annota

**Comment by nmanovic** [comment]
When I open a zip archive with the exported task in annotations directory I see files with '_' suffix (like labels_.json). Why?

**Comment by nmanovic** [comment]
@zhiltsov-max , I don't think that we need annotations in `datumaro` format for now. Thus I will vote for removing `datumaro` and `datumaro_project`. Let's rename `datumaro_project_remote` to  `Datumaro` (or something like that).

**Comment by nmanovic** [comment]
@zhiltsov-max , another comment. Git keeps own config file in '.git' directory. I will suggest keeping config.yaml in .datumaro as well. Users should not modify these files manually in most cases. Better to provide `datum.py config --edit` command instead for advanced users.

File: cvat/apps/engine/views.py [hunk 1]
```diff
@@ -158,10 +158,17 @@ def share(request):
 
     @staticmethod
     @action(detail=False, methods=['GET'], url_path='annotation/formats')
-    def formats(request):
+    def annotation_formats(request):
         data = get_annotation_formats()
         return Response(data)
 
+    @staticmethod
+    @action(detail=False, methods=['GET'], url_path='annotation/dataset_export_formats')
```
Comment (nmanovic) [inline cvat/apps/engine/views.py]:
```suggestion
    @action(detail=False, methods=['GET'], url_path='dataset/formats')
```

What do you think?

File: cvat/apps/engine/views.py [hunk 2]
```diff
@@ -158,10 +158,17 @@ def share(request):
...
+    def annotation_export_formats(request):
```
Comment (nmanovic) [inline cvat/apps/engine/views.py]:
```suggestion
    def dataset_formats(request):
```

File: cvat/apps/dataset_manager/task.py [hunk 3]
```diff
@@ -36,6 +36,9 @@ def log_exception(logger=None, exc_info=True):
 def get_export_cache_dir(db_task):
     return osp.join(db_task.get_task_dirname(), 'export_cache')
 
+EXPORT_FORMAT_DATUMARO_PROJECT = "datumaro_project"
```
Comment (nmanovic) [inline cvat/apps/dataset_manager/task.py]:
```suggestion
EXPORT_FORMAT_DATUMARO_PROJECT = "Dataset repository"
```

Better to have a human readable name here. What do you think?

File: cvat-core/src/server-proxy.js [hunk 4]
```diff
@@ -101,6 +101,22 @@
                 return response.data;
             }
 
+            async function datasetExportFormats() {
+                const { backendAPI } = config;
+
+                let response = null;
+                try {
+                    response = await Axios.get(`${backendAPI}/server/annotation/dataset_export_formats`, {
```
Comment (nmanovic) [inline cvat-core/src/server-proxy.js]:
```suggestion
                    response = await Axios.get(`${backendAPI}/server/dataset/formats`, {
```

File: cvat-core/src/server-proxy.js [hunk 5]
```diff
@@ -223,6 +239,35 @@
                 }
             }
 
+            async function exportDataset(id, format) {
+                const { backendAPI } = config;
+                let url = `${backendAPI}/tasks/${id}/export?format=${format}`;
```
Comment (nmanovic) [inline cvat-core/src/server-proxy.js]: @zhiltsov-max , I would prefer another way:

```suggestion
                let url = `${backendAPI}/tasks/${id}/dataset?format=${format}`;
```

`GET /tasks/${id}/dataset?format=${format}` looks very natural.

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js [hunk 6]
```diff
@@ -207,20 +250,47 @@ class TaskView {
...
         this._annotationFormats = annotationFormats;
+        this._exportFormats = [];
 
+        this._setupExportFormats(exportFormats);
         this._setupList();
         this._setupTaskSearch();
         this._setupCreateDialog();
     }
 
+    _setupExportFormats(availableFormats) {
```
Comment (nmanovic) [inline cvat/apps/dashboard/static/dashboard/js/dashboard.js]: @zhiltsov-max ,

Do you think the server should incapculate the information below? Can we send a json from our server which already has the information?
```
{
    formats: [
        {
            tag: 'datumaro_project',
            name: "Dataset repository",
            is_public: false,
            is_default: true,
        },
        {
            tag: 'coco',
            name: "MS COCO",
            url: "",
            is_public: true,
            is_default: false,
        },
    ], 
}
```

File: cvat/apps/dataset_manager/task.py (hunk 3 above)
Comment (zhiltsov-max) [reply cvat/apps/dataset_manager/task.py]: It is not a name, visible names are declared somewhere in `js`  code.

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js (hunk 6 above)
Comment (zhiltsov-max) [reply cvat/apps/dashboard/static/dashboard/js/dashboard.js]: Technically, I like this idea, but this whole structure looks like client/view logic. What UI pattern do we use?

**Comment by bsekachev** [comment]
@nmanovic 
Probably we need it also in the new UI.

**Pull request merged**